tika-core/src/main/java/org/apache/tika/config/ServiceLoader.java

```diff
@@ -130,7 +130,7 @@ public ServiceLoader(
         this.loader = loader;
         this.handler = handler;
         this.initializableProblemHandler = initializableProblemHandler;
-        this.dynamic = dynamic;
+        this.dynamic = true;
```

Comment:
Isn't this a bit a hacker approach?
This means Tika now will behave differently.. Also this does not solve the issue that in an Osgi enviroment the services might be loaded at different timepoints - so the ServiceLoader might not (yet) know about the service when you try to load the dynamic services (which for e.g. the DefaultEncodingDetector only happens once).